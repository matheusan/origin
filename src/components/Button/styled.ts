import styled from 'styled-components';

export const Button: JSX.Element = styled.button`
  display: block;
  width: ${props => (props.secondary || props.flex ? '100%' : '320px')};
  height: ${props => (props.secondary ? '40px' : '56px')};
  margin: 0 auto;
  border-radius: ${props => (props.secondary ? '8px' : '32px')};
  border: 0;
  outline: 0;
  cursor: pointer;

  background-color: ${props =>
    props.secondary
      ? props.theme.colors.blueGray10
      : props.theme.colors.brandColorPrimary};

  color: ${props =>
    props.secondary
      ? props.theme.colors.brandColorSecondary
      : props.theme.colors.neutralWhite};

  font-family: ${props => props.theme.fonts.primary};
  font-size: 16px;
  line-height: ${props => (props.secondary ? '40px' : '54px')};
  text-align: center;

  &:hover {
    background-color: ${props =>
      props.secondary
        ? props.theme.colors.neutralWhite
        : props.theme.colors.brandColorSecondary};
  }
`;
