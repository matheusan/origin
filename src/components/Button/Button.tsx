import * as React from 'react';

import * as S from './styled';
import Icon from '../Icon';

export interface ButtonProps {
  loading: boolean;
  secondary: boolean;
  onClick?: () => null;
}

const Button: React.SFC<ButtonProps> = ({
  onClick,
  loading = false,
  secondary = false,
  children
}) => (
  <S.Button onClick={onClick} secondary={secondary}>
    {loading ? <Icon name="loading" /> : children}
  </S.Button>
);

export default Button;
