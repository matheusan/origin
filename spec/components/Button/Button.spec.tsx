import * as React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';

import Button from '../../../src/components/Button';

describe('Button', () => {
  const buttonText = 'My button';
  let wrapper: ShallowWrapper;

  beforeEach(() => {
    wrapper = shallow(<Button>{buttonText}</Button>);
  });

  it('renders', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render loading when prop is set', () => {
    wrapper.setProps({ loading: true });
    expect(wrapper.find('Icon').length).toBe(1);
  });
});
