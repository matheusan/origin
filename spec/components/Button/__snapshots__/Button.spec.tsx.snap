// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders 1`] = `
<styled.button
  secondary={false}
>
  My button
</styled.button>
`;

exports[`Button should render seconday color when prop is set 1`] = `
<styled.button
  secondary={true}
>
  My button
</styled.button>
`;
